commands/operator-sdk/cmd/scorecard/test_definitions.go

```diff
@@ -0,0 +1,285 @@
+// Copyright 2019 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scorecard
+
+import (
+	"context"
+
+	olmapiv1alpha1 "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Type Definitions
+
+type Test interface {
+	GetName() string
+	GetDescription() string
+	IsCumulative() bool
```

Comment:
Now that I'm seeing this take shape, I wonder if it's necessary to include `IsCumulative` at this layer of the abstraction.  Since we have concrete implementations of the `Test` interface where `Run()` can return a `TestResult` with any scoring method it wants, should we just leave it up to the concrete Test implementation to return the correct `TestResult`?

Pass/fail tests could accumulate their scores normally and then we could provide a helper function to convert to PassFail:

```go
func ScorePassFail(in *TestResult) *TestResult {
    if in.EarnedPoints != in.MaximumPoints {
        in.EarnedPoints = 0
    }
    return in
}

Reply:
This becomes a problem when we have multiple CRs. Multiple CRs will result in multiple TestResults, which will have to be handled outside of `Run()` (unless each `Run` of a test takes an existing `TestResult` and updates it)